Added the Maven equivalent to the adding-commit-info guide

burtbeckwith opened the pull request.

burtbeckwith committed:
added the Maven equivalent

ilopmar reviewed (commented):
Thanks for this @burtbeckwith :100: 

I've tested it and the `git.properties` file is created when executing the goals `verify`, `package`,... but it's not if I just do `./mvnw mn:run`.

sdelamo commented:
> Thanks for this @burtbeckwith 💯
> 
> I've tested it and the `git.properties` file is created when executing the goals `verify`, `package`,... but it's not if I just do `./mvnw mn:run`.

I think I have seen similar issue for Gradle in the past. If you package the app and run the fat JAR is the git.properties included ?

ilopmar commented:
> I think I have seen similar issue for Gradle in the past. If you package the app and run the fat JAR is the git.properties included ?

Yes

![image](https://user-images.githubusercontent.com/559192/121662142-47537b00-caa5-11eb-9a41-7a619f43a9e9.png)

After that I can even run `./mvnw mn:run` and it works because the file is already there

sdelamo commented on buildSrc/src/main/java/io/micronaut/guides/feature/GitProperties.java:
```diff
@@ -0,0 +1,78 @@
+        String version = resolver.resolve("git-commit-id-plugin")
+                .orElseThrow(() -> new RuntimeException("git-commit-id-plugin not found in pom.xml"))
+                .getVersion();
+        String pluginXml = loadPluginXml().replace("@VERSION@", version);
```
We typically do this simpler. Instead of loading the plugin template and replacing the version we use the `BuildProperties`. See:  

https://github.com/micronaut-projects/micronaut-starter/blob/2.5.x/starter-core/src/main/java/io/micronaut/starter/feature/function/azure/AbstractAzureFunction.java#L92-L98

sdelamo commented on buildSrc/src/main/resources/git-commit-id-plugin.xml:
```diff
@@ -0,0 +1,17 @@
+<plugin>
```
See above comment to replace `@VERSION@` usage.

sdelamo commented on guides/adding-commit-info/adding-commit-info.adoc:
```diff
@@ -56,9 +56,43 @@ plugins {
+
+== Maven Git Commit Id Plugin
+
+If a `git.properties` file is available on the classpath, the http://docs.micronaut.io/latest/api/io/micronaut/management/endpoint/info/source/GitInfoSource.html[GitInfoSource] will expose the values in that file under the `git` key. Generation of a `git.properties` file will need to be configured as part of your build; for example, you may choose to use the https://github.com/git-commit-id/git-c ...
```
Instead of repeating this paragraph: I will move it outside of the build specific section. 

```
If a `git.properties` file is available on the classpath, the http://docs.micronaut.io/latest/api/io/micronaut/management/endpoint/info/source/GitInfoSource.html[GitInfoSource] will expose the values in that file under the `git` key. Generation of a `git.properties` file will need to be configured as part of your build;
```

sdelamo commented on guides/adding-commit-info/adding-commit-info.adoc:
```diff
@@ -56,9 +56,43 @@ plugins {
+
+[source,xml]
+.pom.xml
+----
```
Maybe we should write a macro to load a plugin template into the asciidoc to avoid having it in two places. similar to what we do with `resource`.  thoughts?

burtbeckwith committed:
Merge branch 'master' into git_info_maven

burtbeckwith committed:
delete unused gradle.properties

burtbeckwith committed:
add support for Rocker templates

burtbeckwith committed:
PR feedback

Use BuildProperties instead of string replacement
reduce repetition
use same Rocker template in Asciidoc and in Feature class

burtbeckwith commented:
I configured the plugin to run in the `compile` phase, so it generates the properties file if you run `compile` or another goal that depends on it. `mn:run` does invoke compile, but apparently not in a way that also triggers git.properties generation. I'll wait to resolve this before merging

burtbeckwith committed:
Merge branch 'master' into git_info_maven

sdelamo committed:
Merge branch 'master' into git_info_maven
